housing_cooperative_base/models/res_partner.py

```diff
@@ -79,18 +80,18 @@ class ResPartner(models.Model):
     )
     def _compute_lease_dates(self):
         for partner in self:
-            leases_approved = partner.lease_ids.filtered(
+            approved_leases = partner.lease_ids.filtered(
                 lambda l: l.state != "draft"
-            )
-            partner.first_lease_start = leases_approved.sorted(
-                lambda l: l.start
-            ).start
-            current_lease = leases_approved.filtered(
+            ).sorted(lambda l: l.start)
+            current_leases = approved_leases.filtered(
                 lambda l: l.state == "ongoing"
             )
-            partner.current_lease_id = current_lease
-            partner.current_lease_start = current_lease.start
-            partner.lease_end = current_lease.end
+            if approved_leases:
+                partner.first_lease_start = approved_leases[0].start
+            if current_leases:
```

Comment:
You're assuming that there is only one current lease. It's probably the case for most partners. But I'd choose between these two option:

- use first _current_ lease to have a deterministic result
- use first _current_ lease for start and last _current_ lease for end.

I'd prefer the first one I think.

Reply:
Isn't this what we do given the sort function on line 85?

Reply:
Indeed, missed that. 👌